ports/qemu-riscv/main.c

```diff
@@ -0,0 +1,51 @@
+#include <stdint.h>
```

Comment:
Please add a license header comment.

Reply:
Ok, that is the exact same file as `qemu-arm`'s and it doesn't have a licence header either.  That file was added in 5130b81eecfdd502e8f500f58f39356290f970a3 but then it was worked on by quite a few people.  Whose copyright attribution shall I use for that, the author of the commit I mentioned before?

Reply:
If it's the exact same as `qemu-arm` then I would suggest to remove this file and just refer to the one in `qemu-arm` in the Makefile here.  That way there is no duplication of code, and we can fix the licensing header separately.

Same thing for `test_main.c`, if that's exactly the same.

Reply:
Replying here to the [additional comment below](https://github.com/micropython/micropython/pull/12853/#issuecomment-1803092778) about main.c:

> complicates things on the licensing side as now test_main.c is not the same as qemu-arm's.

Looking through the log of port/qemu-arm/main.c, even back to when it was qemu-arm/main.c, the only contributors are @dpgeorge and @jimmo apart from these commits that all make very small changes in the file: 2face99cfe9 (renames file, no code change), 26a99207 (reorders two headers), 46c3ab20 (two line function signature change), 2fe4cf776 (another function signature change), and d3b1f0b (one line addition, same in every port).

So, I am not a lawyer or hold a copyright on this file, but I reckon it's probably safe to put a generic MicroPython MIT header on this with @dpgeorge's copyright on it. The other contributions are too minor to be copyrightable under most (all?) copyright regimes, and additionally they were all done by long-term contributors who knew they were contributing to an MIT licensed open source project.

> by the way, are there any plans to adopt things like [SPDX](https://spdx.dev/)?

IIRC the CircuitPython folks brought this up recently and Damien is OK with it in principle. (I also think it would be a good change to make!)